tx/src/entities.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+///! This module defines core types that support the transaction processor.
+
+extern crate edn;
+
+// TODO: understand why this is self::edn rather than just edn.
+use self::edn::types::Value;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum EntId {
+    EntId(i64),
+    Ident(String),
```

Comment:
Yeah, fair. That's one reason I split, actually: the 'syntax' of the query parser is the main place I expect `Keyword`s (`:find`, `:in`), with all of the 'data' being namespaced.